gpjax/abstractions.py

```diff
@@ -162,33 +161,44 @@ def fit_batches(
 
     opt_state = optax_optim.init(params)
 
-    prng = convert_seed(seed)
-
-    x, y, n = train_data.X, train_data.y, train_data.n
-
     def loss(params, batch):
         params = trainable_params(params, trainables)
         return objective(params, batch)
 
     @progress_bar_scan(n_iters, log_rate)
     def step(carry, _):
-        params, opt_state, prng = carry
-
-        indicies = jr.choice(prng, n, (batch_size,), replace=True)
+        params, opt_state, current_key = carry
 
-        batch = Dataset(X=x[indicies], y=y[indicies])
+        batch, new_key = get_batch_new_key(train_data, batch_size, current_key)
```

Comment:
I would prefer that the new batch was generated from a specific key i.e., the new key is pased into `get_batch_new_key()`, not the existing key being split within the function. I think for reproducibility, this approach will be cleaner. Unless there's a specific reason for structuring the code as you have done, then I would suggest changing to mimic my suggestion.

Reply:
Instead of having (in the function `get_batch_new_key`) the order:

```python
  indicies = jr.choice(key, n, (batch_size,), replace=True)

   _, new_key = jr.split(key)
```

You would prefer (?):
```python
   key, new_key = jr.split(key)
   indicies = jr.choice(key, n, (batch_size,), replace=True)
```

Is this what you mean?

Reply:
(If so, I'd also prefer this).

Reply:
Yes. In fact, to be consistent with Jax documentation, I'd really like 
```python
 key, subkey = jr.split(key)
 indicies = jr.choice(subkey, n, (batch_size,), replace=True)
```

Reply:
Fantastic! I'll get this done now.

Reply:
Resolved in c31df38